src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8632,6 +8632,18 @@ public override BoundNode VisitEventAssignmentOperator(BoundEventAssignmentOpera
             }
             VisitRvalue(node.Argument);
             // https://github.com/dotnet/roslyn/issues/31018: Check for delegate mismatch.
+            if (node.Argument.ConstantValue?.IsNull != true
+                && MakeMemberSlot(receiverOpt, @event) is not -1 and var memberSlot)
+            {
+                if (node.IsAddition)
+                {
+                    this.State[memberSlot] = this.State[memberSlot].Meet(ResultType.State);
```

Comment:
I believe we should using `SetResult` to set these, rather than updating the state directly like this. Please add some tests that observe the change in field-like event's nullability from the public API.

Reply:
Discussed offline, and the difficulty is that it's not clear what bound node would be passed to SetResult. Will add the test.